sdk/src/main/java/io/customer/sdk/data/store/DeviceStore.kt

```diff
@@ -0,0 +1,72 @@
+package io.customer.sdk.data.store
+
+import android.content.Context
+import android.os.Build
+import io.customer.sdk.Version
+
+interface DeviceStore {
+
+    // Brand: Google
+    val deviceBrand: String
+
+    // Device model: Pixel
+    val deviceModel: String
+
+    // Manufacturer: Samsung
+    val deviceManufacturer: String
+
+    // Android SDK Version: 21
+    val deviceOSVersion: Int
+
+    // Customer App information
+    val customerAppName: String
+    val customerAppVersion: String
+
+    // SDK version
+    val customerIOVersion: String
+
+    fun buildUserAgent(): String
+}
+
+internal class DeviceStoreImp(val context: Context) : DeviceStore {
+
+    private val appInfo: Pair<String, String> = getAppInformation()
+
+    override val deviceBrand: String
+        get() = Build.BRAND
+    override val deviceModel: String
+        get() = Build.MODEL
+    override val deviceManufacturer: String
+        get() = Build.MANUFACTURER
+    override val deviceOSVersion: Int
+        get() = Build.VERSION.SDK_INT
+    override val customerAppName: String
+        get() = appInfo.first
+    override val customerAppVersion: String
+        get() = appInfo.second
+    override val customerIOVersion: String
+        get() = Version.version
+
+    override fun buildUserAgent(): String {
```

Comment:
I know that you're writing automated tests at the moment for the SDK. Friendly reminder to write tests for this function to make sure the generated format is what the backend expects 👍

Reply:
Oh, I suggest a comment [giving the format expected to be generated](https://github.com/customerio/customerio-ios/blob/204176735a8641a77d58232af1daa4e290ee0ca4/Sources/Tracking/Service/HttpClient.swift#L131-L140) by the function. It would be handy for a developer glancing at the function to know the format the API expects to get.